README.md

```diff
@@ -0,0 +1,812 @@
+
+This function should be called right before establishing a TLS/SSL session.
+
+#### process()
+
+Processes any pending data that is received and returns the result code.
+
+When `wait` is set to `false`, the function returns even if the read operation is incomplete.
+
+A wait operation can be interrupted prematurely by returning `true` in the passed predicate.
+
+#### write()
+
+Writes the provided data and returns the result code.
+
+When `wait` is set to `false`, the function returns even if the write operation is incomplete.
+
+A wait operation can be interrupted prematurely by returning `true` in the passed predicate.
+
+### Crypt
+
+Used for encryption & decryption operations.
+
+An object can be constructed through:
+
+- None
+- Move
+
+#### handle()
+
+Returns the OpenSSL handle.
+
+#### cipher()
+
+Returns the cipher name as a read-only string reference ([`std::string_view`](https://en.cppreference.com/w/cpp/string/basic_string_view)).
+
+#### setCipher()
+
+Sets the cipher by name.
+
+Returns whether the operation succeeded.
+
+#### blockSize()
+
+Returns the current cipher's block size.
+
+#### keySize()
+
+Returns the current key's size.
+
+#### nonceSize()
+
+Returns the current nonce's size
+
+#### key()
+
+Returns the current key as a read-only vector of bytes.
+
+#### genKey()
+
+Generates a new key securely and returns it as a vector of bytes.
+
+#### setKey()
+
+Sets the specified key.
+
+Returns whether the operation succeeded.
+
+#### nonce()
+
+Returns the current nonce as a read-only vector of bytes.
+
+#### genNonce()
+
+Generates a new nonce securely and returns it as a vector of bytes.
+
+#### setNonce()
+
+Sets the specified nonce.
+
+Returns whether the operation succeeded.
+
+#### padding()
+
+Returns whether padding is enabled.
+
+#### togglePadding()
+
+Enables or disables padding, depending on the passed flag.
+
+Returns whether the operation succeeded.
+
+#### reset()
+
+Fully resets the OpenSSL context, recommended when rekeying or switching ciphers.
+
+Returns whether the operation succeeded.
+
+#### decrypt()
+
+Decrypts the specified data into the specified buffer.
+
+For AEAD operation you should specify the tag and AAD (additional authenticated data) as well.
+
+Returns the number of bytes written.
+
+#### encrypt()
+
+Encrypts the specified data into the specified buffer.
+
+For AEAD operation you probably want to specify the AAD (additional authenticated data) as well.  
+In that case `tag` should point to a buffer that will receive the produced tag.
+
+Returns the number of bytes written.
+
+### CryptOCB2
+
+This is essentially a special clone of `Crypt` providing support for our OCB2 implementation.
+
+Even if audited, you should rely on `Crypt` instead when possible; it allows to use better ciphers such as ChaCha20-Poly1305 and AES in GCM mode.
+
+For reference, see mumble-voip/mumble#4824.
+
+### Endian
+
+Swaps the bit order on 16, 32 and 64 bits integers.
```

Comment:
```suggestion
Swaps the byte order on 16, 32 and 64 bits integers.
```
The order of bits in individual bytes is unaffected

Reply:
Oops, a lapsus.